torch/_inductor/lowering.py

```diff
@@ -1691,11 +1744,9 @@ def empty_strided(
     dtype = decode_dtype(dtype) or torch.get_default_dtype()
     device = device or torch.tensor(0.0).device
     pointwise = _full(fill_value=0, device=device, dtype=dtype, size=size)
-    if tuple(ir.FlexibleLayout.contiguous_strides(size)) == tuple(stride):
-        # fast path, no need to realize it
-        return pointwise
     pointwise.realize()
     buffer = pointwise.data.data
+    buffer.data.ranges = [0] * len(size)
```

Comment:
How about add a note that we explicitly set the ranges to zeros to mark a noop here?

Reply:
Done.